interview-questions.md

```diff
@@ -7,81 +7,86 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: JavaScript run in html through react.
 
-  Researched answer:
+  Researched answer: It is called JSX, and it is a syntax extension to JavaScript. We recommend using it with React to describe what the UI should look like. JSX may remind you of a template language, but it comes with the full power of JavaScript.
+
+JSX produces React “elements”. We will explore rendering them to the DOM in the next section. Below, you can find the basics of JSX necessary to get you started.
 
 
 
 5. What is React?
 
-  Your answer:
+  Your answer: React is a pre built app that allows small chunks of code to be built and then put together piece by piece.
 
-  Researched answer:
+  Researched answer: React is an open-source JavaScript library for building user interfaces or UI components. It is maintained by Facebook and a community of individual developers and companies. React can be used as a base in the development of single-page or mobile applications.
 
 
 
 6. What is yarn? What file(s) are updated when you run `$ yarn`?
 
-  Your answer:
+  Your answer: yarn is the pre made assests stored inside of React. You get all the files updated that are inside the basic update, both in src and file.
 
-  Researched answer:
+  Researched answer: Yarn is a new package manager that replaces the existing workflow for the npm client or other package managers while remaining compatible with the npm registry. It has the same feature set as existing workflows while operating faster, more securely, and more reliably.
 
 
 
 7. What is localhost:3000? How do we use localhost:3000 in the development process?
 
-  Your answer:
+  Your answer: it is a preinstalled html page that lets us run our React application in real time, though it is not connected to the internet.
 
-  Researched answer:
+  Researched answer: localhost is a network address that points to your own computer. Navigating to localhost:3000 in your web browser sends a request to the Rails development server running locally on port 3000. However, this only works if your web browser is running on the same computer as the development server.
 
 
 
 8. What is an iterable in JavaScript? What data types are iterables? How can this information useful to you as a developer?
 
-  Your answer:
+  Your answer: I honestly do not know what this is and had to look it up.
 
-  Researched answer:
+  Researched answer: The iterable protocol allows JavaScript objects to define or customize their iteration behavior, such as what values are looped over in a for...of construct. Some built-in types are built-in iterables with a default iteration behavior, such as Array or Map , while other types (such as Object ) are not. 
+  - They would be helpful because it is less code to type and more efficiant?
 
 
 
 9. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: Creating a function that can have multiple functions inside it?
```

Comment:
Unfortunately, no ;) But I know your researched answer has it though. Just think of it like this: Your variables need to be declared on the document BEFORE the function that uses them.  If we declare a function that takes in variables that are declared later in our document, it will have to stop what its doing, find those variables later on in the document and HOIST them up to the appropriate place before running. 

Simply put, its bad practice!